allennlp/modules/text_field_embedders/basic_text_field_embedder.py

```diff
@@ -84,7 +85,27 @@ def forward(self, text_field_input: Dict[str, torch.Tensor], num_wrapping_dims:
     # This is some unusual logic, it needs a custom from_params.
     @classmethod
     def from_params(cls, vocab: Vocabulary, params: Params) -> 'BasicTextFieldEmbedder':  # type: ignore
-        # pylint: disable=arguments-differ
+        # pylint: disable=arguments-differ,bad-super-call
+
+        # The original `from_params` for this class was designed in a way that didn't agree
+        # with the constructor. The constructor wants a 'token_embedders' parameter that is a
+        # `Dict[str, TokenEmbedder]`, but the original `from_params` implementation expected those
+        # key-value pairs to be top-level in the params object. In particular, this caused the automatic
+        # FromParams.from_params implementation to do the wrong thing with config files that were
+        # designed for the (idiosyncratic) old behavior.
+        #
+        # This updated implementation delegates to the automatic "from_params" method when it can,
+        # but (for now) it continues to support the old way with a `DeprecationWarning`.
+        if 'token_embedders' in params:
+            # We use `TextFieldEmbedder` as the first argument to super() so that we get
+            # Registrable.from_params.
+            return super(TextFieldEmbedder, cls).from_params(params=params, vocab=vocab)
```

Comment:
Can you just do `super().from_params(...)` here?

Reply:
I could, that would work because `TextFieldEmbedder` itself has no custom `from_params` method and so will delegate further up the inheritance chain to the right method.

however, if someone were to at some point decide we needed a custom implementation for `TextFieldEmbedder.from_params`, then using `super()` would call that custom implementation (and do the wrong thing).

am I being overly defensive? possibly

Reply:
Oh, I missed that this is skipping a class in the hierarchy (skipped over the comment...).  Can't you just be explicit about it and call `FromParams.from_params()`?

Reply:
that doesn't just work, because it's bound to the `Registrable` class and you can't pass in the `BasicTextFieldEmbedder` class to bind to.

however, I figured out how to get it to work, you have to call `__func__` to get the unbound function.

Reply:
I don't have a strong preference about how to do this, I was mostly just curious about what was going on.  Whatever you think is best is fine with me.  If it were me, I'd probably pick just calling `super().from_params(...)`, as I think the argument about something maybe changing above us applies to all of our classes, and we don't worry about it anywhere else.